wgpu-core/src/instance.rs

```diff
@@ -177,10 +166,9 @@ impl Surface {
                 .ok_or(GetSurfacePreferredFormatError::UnsupportedQueueFamily)?
         };
 
-        preferred_formats
-            .iter()
-            .cloned()
-            .find(|preferred| caps.formats.contains(preferred))
+        caps.formats
+            .is_empty()
```

Comment:
This is the inverse condition from what you want - let's just do an early return if statement for the error case

Reply:
Good catch